### PR TITLE
Fix #6 - Deprecate Transport.download(URI, OutputStream, long,IProgressMonitor)

### LINKED ISSUE #6: Deprecate Transport.download(URI, OutputStream, long, IProgressMonitor)
It seems that the method `org.eclipse.equinox.internal.p2.repository.Transport.download(URI, OutputStream, long, IProgressMonitor)` is actually never called from P2.

As `Transport` is an abstract class I suggest to do the following:

1. mark the method deprecated
2. remove abstract modifier and implement it by throwing `OperationNotSupportedException`.

This would make implementing alternative transport much more easier.

### PR DESCRIPTION
**Pull request by laeubi**

### CODE REVIEW

File: bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/Transport.java
Comment (mickaelistria) [inline bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/Transport.java:38]: We need to comment in javadoc with `@deprecated <alternative>` describing that this method isn't used anymore and what other code should be used/implemented instead.

**Review by mickaelistria — CHANGES_REQUESTED** [review]
Just need some more doc.

File: bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/Transport.java
Comment (laeubi) [reply bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/Transport.java:39]: done

**Review by mickaelistria — APPROVED** [review]

**Comment by laeubi** [comment]
@merks @vogella do you like to review?

**Comment by laeubi** [comment]
No need to rush :+1:

**Comment by merks** [comment]
@laeubi 

Yes this is easy to accommodate and I agree that removing functionality that is complex to implement and isn't even used makes sense.

**Comment by laeubi** [comment]
Just approve/merge it when your fine with that :-)

**Comment by vogella** [comment]
LGTM, if you want to have the option to delete the API in 3 years you also need to add the planned deletion to the migration guide.

@akurtakov would it be possible to generate the API to be deleted automatically so that we could get

**Comment by vogella** [comment]
> > if you want to have the option to delete the API in 3 years you also need to add the planned deletion to the migration guide
> 
> This is not API this is an internal class.

Great

**Comment by akurtakov** [comment]
> @akurtakov at some point it be possible to generate the API to be deleted automatically so that we could get rid of this manual step.

Some list is already visible http://help.eclipse.org/2022-03/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Fdeprecated-list.html&anchor=forRemoval not easy to guess when it's fine to remove it.

**Comment by vogella** [comment]
Please merge Jenkins build was succesfull

**Comment by laeubi** [comment]
I'll leave this to @merks (I can't merge it anyways) as he liked to make some additional checks.

**Comment by vogella** [comment]
Looks save to me. @merks any additional concerns? If not I plan to merge early next week.

**Comment by merks** [comment]
As I mentioned in https://github.com/eclipse-equinox/p2/pull/42#issuecomment-1103732728 I'm can implement this in a way that works well for old a new versions of p2.

**Pull request merged**
